### PR TITLE
ndcube Joss paper

### PR DESCRIPTION
**Pull request by DanRyanIrish**
## PR Description

This PR writes a paper for submission to the Journal of Open Source Software.

### CODE REVIEW

**Review by Cadair — APPROVED** [review]

**Comment by DanRyanIrish** [comment]
@Cadair @yashrsharma44 @abit2 @adwaitbhope @nabobalis @hayesla @wtbarnes @BaptistePellorceAstro @rosteen @dhomeier @jmbhughes @ayshih @mwest007

Here is the latest draft of the ndcube JOSS paper which I will submit to be associated with our submission to ApJ.  It justifies why ndcube is needed, summaries the package including its main data classes, and mentions ndcube's broader use.  It is largely a summary of certain parts of the ApJ paper in much less detail.  Please feel free to provide any feedback.

Since we've already submitted the ApJ paper, and since JOSS reviews focus a lot on the code-base, I plan to submit this tomorrow.  If you would like more time to provide feedback, please let me know.

Many thanks!

**Review by dhomeier — CHANGES_REQUESTED** [review]
Thanks, left a few spelling corrections etc. and suggestions on presentation, but I did not go very deep into the technical details.

File: joss_paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+N-dimensional data sets are common in all areas of science and beyond.
+For example, a series of images taken sequentially with a CCD camera can be stored
+as a single 3-D array with two spatial axes and one temporal axis.
+In astronomy, the most commonly used framework for translating between array element
+indices and the location, time, etc. in the Universe being observed is the World
+Coordinate System (WCS).
+WCS’s ability to handle many different physical types (e.g. spatial, temporal,
+spectral, etc.) and projections (e.g. RA and Dec., helioprojective latitude and
+longitude, etc.) make it a succinct, standardized and powerful way to relate array
+axes to the physical types they represent.
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply joss_paper/paper.md:92]: Good point.  I've just made a minor tweak to your suggestion here.

```suggestion
WCS’s ability to handle many different physical types of coordinates (e.g. spatial,
temporal, spectral, etc.) and their projections onto a data array (e.g. right ascension
and declination, helioprojective latitude and longitude, etc.) make it a succinct,
standardized and powerful way to relate array axes to the physical types they represent.
```

File: joss_paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+etc., to minimize friction between analyzing single and multiple cubes.
+
+`NDCollection` allows unordered but related `NDCube` and `NDCubeSequence` objects
+to be linked, similar to how a Python dictionary is used.
+However, in addition to dictionary-like features, `NDCollection` allows axes of
+different cubes with the same lengths to be marked as 'aligned'.
+This enables these axes on all constituent cubes to be sliced at the `NDCollection`
+level.
+One application of this is linking derived data products, e.g. a spectral image cube
+and a Doppler may derived from one of its spectral lines.
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply joss_paper/paper.md:195]:
```suggestion
One application of this is linking derived data products, e.g. a spectral image cube
and a Doppler map derived from one of its spectral lines.
```

**Comment by DanRyanIrish** [comment]
> Thanks, left a few spelling corrections etc. and suggestions on presentation, but I did not go very deep into the technical details.

Thanks @dhomeier!

File: joss_paper/paper.bib [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+@misc{irispy-code,
+  author = {{irispy-lmsal Developers}},
+  title = {irispy-lmsal code-base},
+  year = {2023},
+  publisher = {GitHub},
+  journal = {GitHub Repository},
+  url = {https://github.com/LM-SAL/irispy-lmsal}
+}
+
```
Comment (DanRyanIrish) [inline joss_paper/paper.bib:92]:
```suggestion

@software{jdaviz,
  author       = {JDADF Developers and
                  Averbukh, Jesse and
                  Bradley, Larry and
                  Buikhuizen, Mario and
                  Busko, Ivo and
                  Cherinka, Brian and
                  Conroy, Kyle and
                  Earl, Nicholas and
                  Fox, Ori and
                  Geda, Robel and
                  Jones, Craig and
                  Karatay, Hatice and
                  Kotler, Jenn and
                  Lim, Pey Lian and
                  Morris, Brett and
                  Nguyen, Duy and
                  O'Steen, Richard and
                  Ogaz, Sara and
                  Ogle, Patrick and
                  Otor, O. Justin and
                  Pacifici, Camilla and
                  Robitaille, Thomas and
                  Tollerud, Erik and
                  Volfman, Sabrina},
  title        = {Jdaviz},
  month        = feb,
  year         = 2023,
  note         = {If you use this software, please cite it as below.},
  publisher    = {Zenodo},
  version      = {3.3.0},
  doi          = {10.5281/zenodo.7625637},
  url          = {https://doi.org/10.5281/zenodo.7625637}
}

```

File: joss_paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+The importance of the ndcube package is demonstrated by the fact that it already
+supports a variety of current ground-based and satellite observatories.
+These include the James Webb Space Telescope (JWST), Solar Orbiter,
+the Interface Region Imaging Spectrograph (IRIS), Hinode, and the
+Daniel K. Inouye Solar Telescope (DKIST), as a dependency
+of the specutils [@specutils-docs; @specutils-code], sunraster [@sunraster],
+irispy-lmsal [@irispy-docs; @irispy-code], EISPAC [@eispac-docs; @eispac-code] and
+DKIST user tools packages, respectively.
+It is also used in the data pipeline of the PUNCH mission [Polarimeter to UNify
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:245]:
```suggestion
Daniel K. Inouye Solar Telescope (DKIST) via the
specutils [@specutils-docs; @specutils-code], jdaviz [@jdaviz], sunraster [@sunraster],
irispy-lmsal [@irispy-docs; @irispy-code], EISPAC [@eispac-docs; @eispac-code] and
DKIST user tools packages which all depend on ndcube.
ndcube is also used in the data pipeline of the PUNCH mission [Polarimeter to UNify
```

File: joss_paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+the Corona and Heliosphere; @punch], scheduled for launch in 2025.
+Individual researchers are also using the ndcube package in their own analysis
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:247]:
```suggestion
In addition, individual researchers are using the ndcube package in their own analysis
```

**Review by nabobalis — APPROVED** [review]
LGTM

**Comment by nabobalis** [comment]
Would be nice if it was possible to see the rendered version before submission.

**Comment by Cadair** [comment]
> Would be nice if it was possible to see the rendered version before submission

When you open the PR to JOSS you get the rendered version while they sort out reviewers etc, it's a very laid back process.

**Comment by DanRyanIrish** [comment]
> Would be nice if it was possible to see the rendered version before submission.

Available as a zip file here: https://github.com/DanRyanIrish/ndcube/suites/11451658813/artifacts/590718269

**Comment by DanRyanIrish** [comment]
I have just submitted this to JOSS.  Thanks for all the input!

**Pull request merged**